btx/misc/radial.py

```diff
@@ -111,3 +113,155 @@ def q2pix(qvals, wavelength, distance, pixel_size):
     """
     pix = np.arcsin(qvals*wavelength/2.)
     return 2*distance*pix/pixel_size
+
+class ConcentricCircles:
+
+    def __init__(self, cx, cy, r, num = 100):
+        super().__init__()
+
+        self.cx  = cx                                # Beam center position in pixels along x-axis (axis = 1 in numpy format)
+        self.cy  = cy                                # Beam center position in pixels along y-axis
+        self.r   = np.array([r]).reshape(-1)         # List of radii for all concentric circles in pixels
+        self.num = num                               # Number of pixels sampled from a circle
+        self.crds = np.zeros((2, num * len(self.r))) # Coordinates where pixels are sampled from all circles.  Unit is pixel.  2 is the size of (x, y)
+
+    def generate_crds(self):
+        """
+        Generate coordinates of sample points along each concentric circle
+        """
+        # Fetching initial values that define concentric circles...
+        cx = self.cx
+        cy = self.cy
+        r  = np.array([self.r]).reshape(-1, 1) # Facilitate broadcasting in calculting crds_x and crds_y
+
+        # Find all theta values for generating coordinates of sample points...
+        theta = np.linspace(0.0, 2 * np.pi, self.num)
+        if not isinstance(theta, np.ndarray): theta = np.array([theta]).reshape(-1)
+
+        # Generate coordinates...
+        crds_x = r * np.cos(theta) + cx
+        crds_y = r * np.sin(theta) + cy
+
+        # Reshape crds into one flat array to facilitate optimization routine...
+        self.crds[1] = crds_x.reshape(-1)
+        self.crds[0] = crds_y.reshape(-1)
+
+    def get_pixel_values(self, img):
+        """
+        Get pixel values from all sample points.  If a sample point has 
+        subpixel coordinates, interpolation will take place.  
+
+        Parameters
+        ----------
+        img : numpy.ndarray
+            a powder image
+
+        Returns
+        -------
+        pvals : numpy.ndarray of pixel values.
+            pixel values at all location specified in self.crds        
+        """
+        pvals = map_coordinates(img, self.crds)
```

Comment:
NIT. Same comment as `num`, maybe something more explicit, like `pixel_value` would help read the code.

Reply:
@fredericpoitevin @apeck12 Does `pixel_values` sounds good?  I often use plural to indicate a list or numpy array.  Singular should be fine too.  I am also thinking `pixel_values_from_circles`, but since it's only used in the context of the member function `get_pixel_values` from `ConcentricCircles` class, I guess `pixel_values` is good enough.